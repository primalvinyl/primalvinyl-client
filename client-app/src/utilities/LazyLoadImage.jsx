import React from 'react';
import LazyLoad from 'react-lazyload';
import PropTypes from 'prop-types';

const LazyLoadImage = ({ src, alt, width, height, offset, className }) => {

    return (
        <LazyLoad offset={offset}>
            <img
                src={src}
                alt={alt}
                className={className}
                width={width}
                height={height}
            />
        </LazyLoad>
    );
};

LazyLoadImage.protoTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string,
    className: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
};

LazyLoadImage.defaultProps = {
    src: '',
    alt: '',
    className: undefined,
    width: undefined,
    height: undefined,
};

export default LazyLoadImage;
