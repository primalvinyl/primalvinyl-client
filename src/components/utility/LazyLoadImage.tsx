import React from 'react';
import { Transition } from 'react-transition-group';
import LazyLoad from 'react-lazyload';

const LazyLoadImage = ({
    src,
    alt,
    className,
    offset,
    duration
}: LazyLoadType): React.ReactElement => {

    const [hasLoaded, setHasLoaded] = React.useState(false);
    const containerStyle = {
        transition: `opacity ease-in-out ${duration}ms`,
        opacity: 0,
        width: '100%'
    }
    const imageStyle = {
        maxWidth: '100%',
        height: 'auto'
    };
    const transitionStyle = {
        entering: { opacity: 1 },
        entered: { opacity: 1 },
        exiting: { opacity: 0 },
        exited: { opacity: 0 },
        unmounted: { opacity: 0 }
    };

    return (
        <LazyLoad
            offset={offset}
            classNamePrefix={className}
        >
            <Transition in={hasLoaded} timeout={duration}>
                {state => (
                    <div style={{ ...containerStyle, ...transitionStyle[state] }}>
                        <img
                            width="3000"
                            height="3000"
                            src={src}
                            alt={alt}
                            onLoad={() => setHasLoaded(true)}
                            style={imageStyle}
                        />
                    </div>
                )}
            </Transition>
        </LazyLoad>
    );
};

type LazyLoadType = {
    readonly src: string;
    readonly alt: string;
    readonly className: string;
    readonly offset: number,
    readonly duration: number
};

LazyLoadImage.defaultProps = {
    src: '',
    alt: '',
    className: undefined,
    offset: 150,
    duration: 300
};

export default LazyLoadImage;
