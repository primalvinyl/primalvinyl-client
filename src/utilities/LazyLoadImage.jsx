import React from 'react';
import { Transition } from 'react-transition-group';
import LazyLoad from 'react-lazyload';
import PropTypes from 'prop-types';

const LazyLoadImage = ({
    src,
    alt,
    className,
    offset,
    duration
}) => {

    const [hasLoaded, setHasLoaded] = React.useState(false);
    const containerStyle = {
        transition: `opacity ease-in-out ${duration}ms`,
        opacity: 0,
        width: '100%'
    }
    const imageStyle = {
        maxWidth: '100%',
        height: 'auto'
    };
    const transitionStyle = {
        entering: { opacity: 1 },
        entered: { opacity: 1 },
        exiting: { opacity: 0 },
        exited: { opacity: 0 },
    };

    return (
        <LazyLoad
            offset={offset}
            classNamePrefix={className}
        >
            <Transition in={hasLoaded} timeout={duration}>
                {state => (
                    <div style={{ ...containerStyle, ...transitionStyle[state] }}>
                        <img
                            width="3000"
                            height="3000"
                            src={src}
                            alt={alt}
                            onLoad={() => setHasLoaded(true)}
                            style={imageStyle}
                        />
                    </div>
                )}
            </Transition>
        </LazyLoad>
    );
};

LazyLoadImage.protoTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string,
    className: PropTypes.string,
    offset: PropTypes.number,
    duration: PropTypes.number
};

LazyLoadImage.defaultProps = {
    alt: '',
    className: undefined,
    offset: 150,
    duration: 300
};

export default LazyLoadImage;
