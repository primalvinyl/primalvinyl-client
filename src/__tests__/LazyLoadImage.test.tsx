import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import LazyLoadImage from '../components/utility/LazyLoadImage';

describe('Header', () => {
    it('renders', () => {
        const myWrapper = shallow(
            <LazyLoadImage />
        );
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(
            <LazyLoadImage />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
