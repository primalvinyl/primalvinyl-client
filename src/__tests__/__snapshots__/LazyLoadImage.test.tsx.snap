// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  className="lazyload-wrapper"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="lazyload-placeholder"
    style={
      Object {
        "height": undefined,
      }
    }
  />
</div>
`;
